components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -3045,6 +2888,425 @@ subroutine check_values(Qv,T,i,timestepcount,force_abort,source_ind)
     endif
 
   end subroutine check_values
-  !===========================================================================================
+
+  subroutine ice_droplet_collection(rho,t,rhofaci,    &
+  f1pr04,qitot_incld,qc_incld,nitot_incld,nc_incld,    &
+             qccol,nccol,qcshd,ncshdc)
+   
+   !.......................
+   ! collection of droplets
+
+   ! here we multiply rates by air density, air density fallspeed correction
+   ! factor, and collection efficiency since these parameters are not
+   ! included in lookup table calculations
+   ! for T < 273.15, assume collected cloud water is instantly frozen
+   ! note 'f1pr' values are normalized, so we need to multiply by N
+
+
+   implicit none 
+
+   real(rtype), intent(in) :: rho
+   real(rtype), intent(in) :: t 
+   real(rtype), intent(in) :: rhofaci 
+   real(rtype), intent(in) :: f1pr04
```

Comment:
We probably want to keep the variable names the same in and out of sub-routines.  But we may want to consider adding a descriptor for these opaque table lookup variables so that it is easier to figure out what is going on.  i.e. 
f1pr04 is `! collection of cloud water by ice`